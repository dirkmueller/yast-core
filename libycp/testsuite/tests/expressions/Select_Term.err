Parsed:
----------------------------------------------------------------------
{
    // term node1
    // term node2
    // term node3
    // term node4
    // term node5
    // term node6
    // term node7
    // filename: "tests/expressions/Select_Term.ycp"
    term node1 = `Termified (`Austria, `Belgium);
    term node2 = `Termified (`China, `D_Nation);
    term node3 = `Termified (`England, `France);
    term node4 = `Termified (node2, node3);
    term node5 = `Termified (node1, node4);
    term node6 = `Termified (`Greece, nil);
    term node7 = `Termified (node5, node6);
    return (select (node1, 0, nil) == `Austria);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // term node1
    // term node2
    // term node3
    // term node4
    // term node5
    // term node6
    // term node7
    // filename: "tests/expressions/Select_Term.ycp"
    term node1 = `Termified (`Austria, `Belgium);
    term node2 = `Termified (`China, `D_Nation);
    term node3 = `Termified (`England, `France);
    term node4 = `Termified (node2, node3);
    term node5 = `Termified (node1, node4);
    term node6 = `Termified (`Greece, nil);
    term node7 = `Termified (node5, node6);
    return (select (node2, 0, nil) == `China);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // term node1
    // term node2
    // term node3
    // term node4
    // term node5
    // term node6
    // term node7
    // filename: "tests/expressions/Select_Term.ycp"
    term node1 = `Termified (`Austria, `Belgium);
    term node2 = `Termified (`China, `D_Nation);
    term node3 = `Termified (`England, `France);
    term node4 = `Termified (node2, node3);
    term node5 = `Termified (node1, node4);
    term node6 = `Termified (`Greece, nil);
    term node7 = `Termified (node5, node6);
    return (node6[1]:nil == nil);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // term node1
    // term node2
    // term node3
    // term node4
    // term node5
    // term node6
    // term node7
    // filename: "tests/expressions/Select_Term.ycp"
    term node1 = `Termified (`Austria, `Belgium);
    term node2 = `Termified (`China, `D_Nation);
    term node3 = `Termified (`England, `France);
    term node4 = `Termified (node2, node3);
    term node5 = `Termified (node1, node4);
    term node6 = `Termified (`Greece, nil);
    term node7 = `Termified (node5, node6);
    return (/* any -> term */node7[0, 1, 0]:`error () == node2);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // term node1
    // term node2
    // term node3
    // term node4
    // term node5
    // term node6
    // term node7
    // filename: "tests/expressions/Select_Term.ycp"
    term node1 = `Termified (`Austria, `Belgium);
    term node2 = `Termified (`China, `D_Nation);
    term node3 = `Termified (`England, `France);
    term node4 = `Termified (node2, node3);
    term node5 = `Termified (node1, node4);
    term node6 = `Termified (`Greece, nil);
    term node7 = `Termified (node5, node6);
    return (/* any -> const symbol */node7[0, 1, 1, 1]:`error == `France);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // term node1
    // term node2
    // term node3
    // term node4
    // term node5
    // term node6
    // term node7
    // filename: "tests/expressions/Select_Term.ycp"
    term node1 = `Termified (`Austria, `Belgium);
    term node2 = `Termified (`China, `D_Nation);
    term node3 = `Termified (`England, `France);
    term node4 = `Termified (node2, node3);
    term node5 = `Termified (node1, node4);
    term node6 = `Termified (`Greece, nil);
    term node7 = `Termified (node5, node6);
    return node7;
}
----------------------------------------------------------------------
