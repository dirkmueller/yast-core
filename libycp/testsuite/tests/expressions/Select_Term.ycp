
# ---------------------------------------------------------
#
#  Filename:    Select_Term.ycp
#
#  Purpose:     test cases for YCP 'select' on terms
#
#  Creator:     jd@suse.de
#
#  Maintainer:  jd@suse.de
#
# ---------------------------------------------------------

{
    term  node1 = `Termified( `Austria, `Belgium );
    term  node2 = `Termified( `China, `D_Nation );
    term  node3 = `Termified( `England, `France );
    term  node4 = `Termified( node2, node3 );
    term  node5 = `Termified( node1, node4 );
    term  node6 = `Termified( `Greece, nil );
    term  node7 = `Termified( node5, node6 );

    return( select( node1, 0, nil) == `Austria );
}

{
    term  node1 = `Termified( `Austria, `Belgium );
    term  node2 = `Termified( `China, `D_Nation );
    term  node3 = `Termified( `England, `France );
    term  node4 = `Termified( node2, node3 );
    term  node5 = `Termified( node1, node4 );
    term  node6 = `Termified( `Greece, nil );
    term  node7 = `Termified( node5, node6 );

    return( select( node2, 0, nil) == `China );
}

{
    term  node1 = `Termified( `Austria, `Belgium );
    term  node2 = `Termified( `China, `D_Nation );
    term  node3 = `Termified( `England, `France );
    term  node4 = `Termified( node2, node3 );
    term  node5 = `Termified( node1, node4 );
    term  node6 = `Termified( `Greece, nil );
    term  node7 = `Termified( node5, node6 );

    return( node6[1]:nil == nil );
}

{
    term  node1 = `Termified( `Austria, `Belgium );
    term  node2 = `Termified( `China, `D_Nation );
    term  node3 = `Termified( `England, `France );
    term  node4 = `Termified( node2, node3 );
    term  node5 = `Termified( node1, node4 );
    term  node6 = `Termified( `Greece, nil );
    term  node7 = `Termified( node5, node6 );

    return( node7[0, 1, 0]:`error() == node2 );
}

{
    term  node1 = `Termified( `Austria, `Belgium );
    term  node2 = `Termified( `China, `D_Nation );
    term  node3 = `Termified( `England, `France );
    term  node4 = `Termified( node2, node3 );
    term  node5 = `Termified( node1, node4 );
    term  node6 = `Termified( `Greece, nil );
    term  node7 = `Termified( node5, node6 );

    return( node7[0, 1, 1, 1]:`error == `France );
}

{
    term  node1 = `Termified( `Austria, `Belgium );
    term  node2 = `Termified( `China, `D_Nation );
    term  node3 = `Termified( `England, `France );
    term  node4 = `Termified( node2, node3 );
    term  node5 = `Termified( node1, node4 );
    term  node6 = `Termified( `Greece, nil );
    term  node7 = `Termified( node5, node6 );

    return( node7 );
}

