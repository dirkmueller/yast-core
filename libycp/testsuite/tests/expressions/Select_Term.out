(true)
(true)
(true)
(true)
(true)
(`Termified (`Termified (`Termified (`Austria, `Belgium), `Termified (`Termified (`China, `D_Nation), `Termified (`England, `France))), `Termified (`Greece, nil)))
